### DISCUSSION TITLE
Burnside groups with GAP system

### DISCUSSION THREAD

**Discussion by ArthurBesse** [gap-system/gap#5696] [Q&A] [answered]
My question is related to Burnside groups B(n, 3) in the GAP system. I'm interested in ways to represent Burnside groups B(n, 3) in GAP. The obvious representation using relations (see example for B(3,3) below) is quite complex, because obtaining relations for large n is not an easy task.

```
f := FreeGroup(3);;
a := f.1;;
b := f.2;;
c := f.3;;
rels := [a^3, b^3, c^3, (a*b)^3, (a*c)^3, (b*c)^3, (a^2*b)^3, (a^2*c)^3, (b^2*c)^3, (a*b*c)^3, (a^2*b*c)^3, (a*b^2*c)^3, (b^2*a^2*c)^3];;
g := f / rels;;
```
1. Is there a convenient way in GAP (maybe using some external package) to represent relatively free groups, in particular Burnside groups, in particular B(n, 3)
?
2. Is there a convenient way in GAP (maybe using some external package) to represent verbal subgroups (this can be used to simplify the method with relations mentioned above)?

Thanks in advance.

**Comment by fingolfin** [comment, marked as answer]
Ugh, somehow I never saw this thread (we are not using these "discussions" much, we usually use [Slack](https://gap-system.org/slack) or our mailing lists instead).

But to answer your question, just in case you or some other future visitor still care: the `anupq` package can define Burnside groups and similar verbal groups. For infinite nilpotent groups one can use the `nq` package.

Here is one example taken from [the anupq manual](https://gap-packages.github.io/anupq/doc/chap4_mj.html) which you can adapt to compute $B(n,3)$:
```gap-repl
gap> F := FreeGroup(2);
<free group on the generators [ f1, f2 ]>
gap> Burnside5 := x->x^5;
function( x ) ... end
gap> Engel3 := function( x,y ) return PqLeftNormComm( [x,y,y,y] ); end;
function( x, y ) ... end
gap> Pq( F : Prime := 5, Identities := [ Burnside5, Engel3 ] );
#I  Class 1 with 2 generators.
#I  Class 2 with 3 generators.
#I  Class 3 with 5 generators.
#I  Class 3 with 5 generators.
<pc group of size 3125 with 5 generators>
```

So adjusted we get:
```gap-repl
gap> G := Pq( FreeGroup(2) : Prime := 3, Identities := [ x -> x^3 ] );
#I  Class 1 with 2 generators.
#I  Class 2 with 3 generators.
#I  Class 2 with 3 generators.
<pc group of size 27 with 3 generators>
gap> G := Pq( FreeGroup(3) : Prime := 3, Identities := [ x -> x^3 ] );
#I  Class 1 with 3 generators.
#I  Class 2 with 6 generators.
#I  Class 3 with 7 generators.
#I  Class 3 with 7 generators.
<pc group of size 2187 with 7 generators>
gap> G := Pq( FreeGroup(4) : Prime := 3, Identities := [ x -> x^3 ] );
#I  Class 1 with 4 generators.
#I  Class 2 with 10 generators.
#I  Class 3 with 14 generators.
#I  Class 3 with 14 generators.
<pc group of size 4782969 with 14 generators>
```

**Reply by ArthurBesse** [reply]
Thank you very much!
